system-x/services/cert-manager/src/main/java/software/tnb/certmanager/validation/CertManagerValidation.java

```diff
@@ -0,0 +1,156 @@
+package software.tnb.certmanager.validation;
+
+import software.tnb.common.openshift.OpenshiftClient;
+import software.tnb.common.validation.Validation;
+
+import org.apache.velocity.Template;
+import org.apache.velocity.VelocityContext;
+import org.apache.velocity.app.VelocityEngine;
+import org.apache.velocity.runtime.RuntimeConstants;
+import org.apache.velocity.runtime.resource.loader.ClasspathResourceLoader;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import io.fabric8.kubernetes.api.model.GenericKubernetesResourceBuilder;
+import io.fabric8.kubernetes.api.model.ServiceAccount;
+import io.fabric8.kubernetes.api.model.ServiceAccountBuilder;
+import io.fabric8.kubernetes.api.model.rbac.PolicyRule;
+import io.fabric8.kubernetes.api.model.rbac.Role;
+import io.fabric8.kubernetes.api.model.rbac.RoleBinding;
+import io.fabric8.kubernetes.api.model.rbac.RoleBindingBuilder;
+import io.fabric8.kubernetes.api.model.rbac.RoleBuilder;
+import io.fabric8.kubernetes.api.model.rbac.RoleRef;
+import io.fabric8.kubernetes.api.model.rbac.Subject;
+import io.fabric8.kubernetes.client.dsl.base.CustomResourceDefinitionContext;
+
+public class CertManagerValidation implements Validation {
+
+    private static final CustomResourceDefinitionContext ISSUER_CTX = new CustomResourceDefinitionContext
+        .Builder()
+        .withGroup("cert-manager.io")
+        .withKind("Issuer")
+        .withName("issuers.cert-manager.io")
+        .withPlural("issuers")
+        .withScope("Namespaced")
+        .withVersion("v1")
+        .build();
+
+    private static final CustomResourceDefinitionContext CERTIFICATE_CTX = new CustomResourceDefinitionContext
+        .Builder()
+        .withGroup("cert-manager.io")
+        .withKind("Certificate")
+        .withName("certificates.cert-manager.io")
+        .withPlural("certificates")
+        .withScope("Namespaced")
+        .withVersion("v1")
+        .build();
+
+    /**
+     * Creates self-signed issuer in the current namespace
+     */
+    public void createSelfSignedIssuer() {
+        OpenshiftClient.get().genericKubernetesResources(ISSUER_CTX)
+            .inNamespace(OpenshiftClient.get().getNamespace())
+            .resource(new GenericKubernetesResourceBuilder()
+                .withKind(ISSUER_CTX.getKind())
+                .withNewMetadata()
+                .withName("selfsigned-issuer")
+                .endMetadata()
+                .withAdditionalProperties(Map.of("spec", Map.of(
+                    "selfSigned", Map.of()
+                )))
+                .build()
+            ).create();
```

Comment:
potentially the same problem as with `createSecretViewer`

Reply:
changed to `.serverSideApply()` in https://github.com/tnb-software/TNB/pull/1291